pkg/controller/common/util.go

```diff
@@ -1803,7 +1803,7 @@ func ValidateSnapshotCloneProvisioners(ctx context.Context, c client.Client, sna
 }
 
 // GetSnapshotClassForSmartClone looks up the snapshot class based on the storage class
-func GetSnapshotClassForSmartClone(dvName string, targetPvcStorageClassName *string, log logr.Logger, client client.Client) (string, error) {
+func GetSnapshotClassForSmartClone(dvName string, targetPvcStorageClassName, snapshotClassName *string, log logr.Logger, client client.Client) (string, error) {
```

Comment:
Am I missing something, or does this need to change too?
https://github.com/kubevirt/containerized-data-importer/blob/581dfb91c7699a587335ff63fe1120958a776886/pkg/controller/clone/common.go#L214-L215

Reply:
Yes it should, and I guess they should be refactored to reuse the same code.

Reply:
I can see how its more tricky in the populator case, since you probably don't want to start looking up storage profiles here

Reply:
I am more concerned about the coupling of storage profiles and our populators
@alromeros Did we ever mean for our set of populators to be used standalone?
Or is this fine to look up storage profiles?

Reply:
> @alromeros Did we ever mean for our set of populators to be used standalone?

Yeah... so it depends on what we mean by standalone usage. We first merged the populators without DV support and ensured it was well-tested and reliable as a standalone feature. However they'll always be dependent on CDI controllers, so don't know if relying on storage profiles would be against this design decision.

Reply:
Yeah, we plan to keep depending on all import, upload, and clone controllers. So I guess that if the main rationale behind populators was always to stop depending on DVs (and we are close to achieving that), there's no problem with resorting to storage profiles since they were never an issue to begin with.